src/main/telemetry/smartport.c

```diff
@@ -160,30 +169,97 @@ static uint8_t smartPortHasRequest = 0;
 static uint8_t smartPortIdCnt = 0;
 static uint32_t smartPortLastRequestTime = 0;
 
+typedef struct smartPortFrame_s {
+    uint8_t  sensorId;
+    uint8_t  frameId;
+    uint16_t valueId;
+    uint32_t data;
+} smartPortFrame_t;
+
+#define SMARTPORT_FRAME_SIZE  sizeof(smartPortFrame_t)
+#define SMARTPORT_TX_BUF_SIZE 256
+
+#define SMARTPORT_PAYLOAD_SIZE   (SMARTPORT_FRAME_SIZE - 2*sizeof(uint8_t))
+#define SMARTPORT_PAYLOAD_OFFSET 2
+
+static uint8_t smartPortRxBuffer[SMARTPORT_FRAME_SIZE];
+static uint8_t smartPortRxBytes = 0;
+static bool smartPortFrameReceived = false;
+
+#define SMARTPORT_MSP_VERSION    1
```

Comment:
Well, I'd prefer just using another frame ID for other usage, and keep these bits for MSP-related things.